java/client/src/test/java/io/vitess/client/RpcClientTest.java

```diff
@@ -144,6 +150,45 @@ public void setUp() {
     return values;
   }
 
+  /**
+   * waitForVtgateclienttest waits up to 60 seconds for the "vtgateclienttest" binary to come up
+   * and be reachable on the RPC port.
+   *
+   * We use the "GetSrvKeyspace" RPC exemplary to test if vtgate responds.
+   *
+   * @throws SQLException
+   * @throws InterruptedException
+   */
+  private void waitForVtgateclienttest() throws SQLException, InterruptedException {
+    DateTime start = DateTime.now();
+    DateTime deadline = start.plusSeconds(60);
+
+    boolean waited = false;
+    while (DateTime.now().isBefore(deadline)) {
+      try {
+        ctx = Context.getDefault().withDeadlineAfter(Duration.millis(5000)).withCallerId(CALLER_ID);
```

Comment:
Does this actually work?  I remember having a conversation with @harshit-gangal back in December, about this I believe.  If I recall correctly, I tried doing something similar to this... but discovered that the next level down in the stack is dropping these `Context` objects on the floor and not doing anything with them.

See the discussion between Harshit and myself in this PR (apologies for its size), about `GrpcClientFactory.java` on Dec. 28:

https://github.com/youtube/vitess/pull/2349

https://reviewable.io/reviews/youtube/vitess/2349#-K_5emU-Ea0R993KCDA-